core/src/components/checkbox/checkbox.tsx

```diff
@@ -245,6 +259,7 @@ export class Checkbox implements ComponentInterface {
             onBlur={() => this.onBlur()}
             ref={(focusEl) => (this.focusEl = focusEl)}
           />
+          {mode === 'md' && inItem && <ion-ripple-effect></ion-ripple-effect>}
```

Comment:
IMO we should always render this in MD mode that way developers can use the ripple effect outside of `ion-item` if they want to

Reply:
Checkboxes don't currently have a ripple effect with the legacy syntax, so I'm hesitant to change that. The [MD spec](https://m2.material.io/develop/web/components/input-controls/checkboxes) does show some different-looking hover/active/focus states, with the circle wrapping around the checkbox, but we don't have any of that in `main` either so making the change feels out of scope here.